cli/cmdlineparser.cpp

```diff
@@ -1022,6 +1026,11 @@ bool CmdLineParser::parseFromArgs(int argc, const char* const argv[])
     if (mSettings->basePaths.empty() && mSettings->relativePaths)
         mSettings->basePaths = mPathNames;
 
+    if (mSettings->xml && mSettings->sarif) {
+        printError("output format cannot be both XML and SARIF.");
```

Comment:
I agree that `--output-format` makes sense. Then you could replace `--xml`, `--xml-version`, `--sarif`, and potentially even `--sarif-version` (not implemented, but could exist depending on future versions of the standard) with `--output-format {xml|xml2|xml3|sarif|sarif-2.1.0}`.

Reply:
However, for the sake of keeping this PR small and isolated, I think that converting `--xml`/`--sarif` to `--output-format` should be a separate PR.

>Maybe clarify which options need to be exclusive instead of using the acronyms.

Fair enough.

Reply:
We need to keep `--xml` for a while anyway. For a while it would be possible to write both `--xml` and `--output-format=xml`. But the `--sarif` is not needed.

Reply:
> We need to keep `--xml` for a while anyway. For a while it would be possible to write both `--xml` and `--output-format=xml`. But the `--sarif` is not needed.

+1

Reply:
Done.